hack/kind-cluster-build.sh

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env bash
+set -e
+
+usage() {
+    cat <<EOF
+This script use kind to create Kubernetes cluster,about kind please refer: https://kind.sigs.k8s.io/
+Before run this script,please ensure that:
+* have installed git (version >= v2.15.1)
+* have installed docker
+
+Options:
+       -n,--name               Name of the Kubernete cluster,default value: kind
+       -c,--nodeNum            the count of the cluster nodes,default value: 6
+       -v,--k8sVersion        version of the Kubernetes cluster,default value: v1.12.8
+Usage:
+    $0 --name testCluster --nodeNum 4 --k8sVersion v1.12.9
+EOF
+    exit 1
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    -n|--name)
+    clusterName="$2"
+    shift
+    shift
+    ;;
+    -c|--nodeNum)
+    nodeNum="$2"
+    shift
+    shift
+    ;;
+    -v|--k8sVersion)
+    k8sVersion="$2"
+    shift
+    shift
+    ;;
+    *)
+    echo "unknown option: $key"
+    usage
+    exit 1
+    ;;
+esac
+done
+
+clusterName=${clusterName:-kind}
+nodeNum=${nodeNum:-6}
+k8sVersion=${k8sVersion:-v1.12.8}
+
+echo "clusterName: ${clusterName}"
+echo "nodeNum: ${nodeNum}"
+echo "k8sVersion: ${k8sVersion}"
+
+echo "############ check kind ##############"
+if hash kind 2>/dev/null;then
+    echo "kind have installed"
+else
+    echo "this script needs kind, please install kind first."
+    ehco "you can refer: https://kind.sigs.k8s.io/"
+    exit 1
+fi
+
+echo "############ check kubectl##############"
+if hash kubectl 2>/dev/null;then
+    echo "kubectl have installed"
+else
+    echo "this script needs kubectl, please install kubectl first."
+    exit 1
+fi
+
+echo "############ check helm ##############"
+if hash helm 2>/dev/null;then
+    echo "helm have installed"
+else
+    echo "this script needs helm, please install helm first"
+    exit 1
+fi
+
+echo "############ check docker ##############"
+if hash docker 2>/dev/null;then
+    echo "docker have installed"
+else
+    echo "this script needs docker, please install docker first"
+    exit 1
+fi
+
+echo "############# start create cluster:[${clusterName}] #############"
+
+data_dir=${HOME}/kind/${clusterName}/data
+
+echo "clean data dir: ${data_dir}"
+if [ -d ${data_dir} ]; then
+    rm -rf ${data_dir}
+fi
+
+echo "init the mount dirs for kind cluster"
+maxindex=$[nodeNum-1]
+mkdir -p ${data_dir}/worker{0..${maxindex}}/vol{1..9}
+
+configFile=${HOME}/kind/${clusterName}/kind-config.yaml
+
+cat <<EOF > ${configFile}
+kind: Cluster
+apiVersion: kind.sigs.k8s.io/v1alpha3
+networking:
+  disableDefaultCNI: true
+nodes:
+- role: control-plane
+EOF
+
+for ((i=0;i<=${maxindex};i++))
+do
+cat <<EOF >>  ${configFile}
+- role: worker
+  extraMounts:
+  - containerPath: /mnt/disks/vol1
```

Comment:
Could you make the volume count be configurable and avoid the following verbose path mount.